docs/guide/advanced/transitions.md

```diff
@@ -1 +1,53 @@
 # Transitions
+
+In general, you may want to test the resulting DOM after a transition, and this is why Vue Test Utils mocks `<transition>` and `<transition-group>` by default.
+
+Following is a simple component that toggles a content wrapped in a fading transition:
+
+```vue
+<template>
+  <button @click="show = !show">Toggle</button>
+
+  <transition name="fade">
+    <p v-if="show">hello</p>
+  </transition>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      show: false
+    }
+  }
+}
+</script>
```

Comment:
Maybe we could use a more modern way of defining components in the docs?
Something like:

```vue
<script setup>
import { ref } from 'vue'

const show = ref(false)
</script>
```

Reply:
that's a good debate – we've been using SFC or the "old way" of defining components throughout the docs where needed, but I also feel we should demonstrate VTU works fine with `script setup` and refs and whatnot.

Do you think we should use arbitrary ways of defining components? Or maybe we should write a specific page for `script setup`?

pinging @lmiller1990 @xanf to share their thoughts :)

Reply:
Yes, it's hard to know what would be best. I would at least not use `data` and replace it with a setup function if we don't want to use `script setup` directly.

Reply:
OK 👌  so for the time being I'll replace it with a `setup` function, but let's keep on exploring the best way to showcase that VTU works with any combination of options API/options API with setup/`<script setup>`

Reply:
I think we should pick 1 style and standardize on in.

I'd say "just do what Vue core docs are doing" but those are all over the shop at the moment. It feels weird to push TS and script setup (although this is my favorite way to code Vue now) if the docs for other core libs are not doing the same.

From what I can see, we are using the options API mostly in the Vue next docs - I think we should use that here (for now) to be consistent. I'd like to move towards TS and script setup, but we should do this in one single go.

Reply:
I guess options API + `setup()` were sensible is a good default?

Reply:
Seems fine - I'd like to actually move everything to TS and script setup soonish, but we should discuss this first, and do it all in one go.

Reply:
great, then we're good to go! Already changed this page to `setup()`